SpringSecurityOauth2ProviderGrailsPlugin.groovy

```diff
@@ -470,6 +475,26 @@ class SpringSecurityOauth2ProviderGrailsPlugin {
         }
     }
 
+    private configureBasicAuthenticationFilter = { conf ->
+        // Should the basic authentication filter be registered in the filter chain
+        boolean registerBasicAuthFilter = conf.oauthProvider.registerBasicAuthFilter as boolean
+
+        if(registerBasicAuthFilter) {
+            // This defines the bean using the alternative constructor that sets ignoreFailure to true,
+            // as an alternative to the filter offered by the Spring Security Core plugin:
+            basicAuthenticationFilter(BasicAuthenticationFilter, ref('authenticationManager', ref('oauth2AuthenticationEntryPoint'))) {
+                authenticationDetailsSource = ref('authenticationDetailsSource')

```

Comment:
Similar to the above, we should provide a handle to an instance of `OAuth2AuthenticationDetailsSource` to avoid any conflicts and because it provides OAuth2 specific details.